x-pack/plugins/ml/public/data_frame/components/source_index_preview/common.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dictionary } from '../../../../common/types/common';
+
+export type EsFieldName = string;
+
+type EsId = string;
+type EsDocSource = Dictionary<any>;
+
+export interface EsDoc extends Dictionary<any> {
+  _id: EsId;
+  _source: EsDocSource;
+}
+
+export const MAX_COLUMNS = 5;
+
+function getFlattenedFields(obj: EsDocSource): EsFieldName[] {
+  const flatDocFields: EsFieldName[] = [];
+  const newDocFields = Object.keys(obj);
+  newDocFields.forEach(f => {
+    const fieldValue = obj[f];
+    if (typeof fieldValue !== 'object' || fieldValue === null || Array.isArray(fieldValue)) {
```

Comment:
Not quite understanding here why we would want to push in the `field` if it's value is equal to `null`. Are we just trying to get all valid field names even if they are not populated?

Reply:
Yes it's just about getting the field. Because I'm taking just the first document of the result set, it could just be that in the case of that first document we're seeing `null` and other docs might be populated with other values.